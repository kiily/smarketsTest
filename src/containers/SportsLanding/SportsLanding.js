import React, { PureComponent, Fragment } from 'react';
import PropTypes from 'prop-types';

import EventInfoList from '../../components/EventInfoList/EventInfoList';
import ContentHeader from '../../components/ContentHeader/ContentHeader';
import Spinner from '../../components/Spinner/Spinner';

class SportsLanding extends PureComponent {

  render() {
    console.log('sporttsLanding', this.props);
    return (
      <Fragment>
        <ContentHeader currentSport={this.props.currentSport}/>
        {this.props.eventsLoaded ? <EventInfoList events={this.props.popularEvents}/> : <Spinner/>}
      </Fragment>
    );
  }
}

SportsLanding.propTypes = {
  currentSport: PropTypes.string,
  eventsLoaded: PropTypes.bool,
  popularEvents: PropTypes.array
};

export default SportsLanding;
