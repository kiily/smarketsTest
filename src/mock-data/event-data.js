export const events = [
  {
    'actual_start_datetime': null,
    'bettable': true,
    'created': '2019-04-16T10:09:18.291988Z',
    'description': null,
    'display_order': 0,
    'end_date': null,
    'full_slug': '/sport/football/championship/2019/04/30/millwall-fc-vs-bristol-city-fc',
    'hidden': false,
    'parent': {
      'name': 'Championship'
    },
    'id': '1925896',
    'inplay_enabled': true,
    'modified': '2019-04-16T10:09:18.291988Z',
    'name': 'Millwall vs. Bristol City',
    'parent_id': '939285',
    'short_name': 'MWL vs. BRC',
    'slug': 'millwall-fc-bristol-city-fc-2019-04-30',
    'special_rules': null,
    'start_date': '2019-04-30',
    'start_datetime': '2019-04-30T18:45:00Z',
    'state': 'upcoming',
    'type': 'football_match'
  },
  {
    'actual_start_datetime': null,
    'bettable': true,
    'created': '2019-04-17T10:10:36.025237Z',
    'description': null,
    'display_order': 0,
    'end_date': null,
    'full_slug': '/sport/football/championship/2019/05/01/swansea-vs-derby',
    'hidden': false,
    'parent': {
      'name': 'Championship'
    },
    'id': '1934371',
    'inplay_enabled': true,
    'modified': '2019-04-17T10:10:36.025237Z',
    'name': 'Swansea City vs. Derby County',
    'parent_id': '939285',
    'short_name': 'SWA vs. DER',
    'slug': 'swansea-derby-2019-05-01',
    'special_rules': null,
    'start_date': '2019-05-01',
    'start_datetime': '2019-05-01T18:45:00Z',
    'state': 'upcoming',
    'type': 'football_match'
  },
  {
    'actual_start_datetime': null,
    'bettable': true,
    'created': '2019-04-18T10:11:56.786364Z',
    'description': null,
    'display_order': 0,
    'end_date': null,
    'full_slug': '/sport/football/europe-uefa-champions-league/2019/04/30/tottenham-vs-afc-ajax',
    'hidden': false,
    'parent': {
      'name': 'Champions League'
    },
    'id': '1943310',
    'inplay_enabled': true,
    'modified': '2019-04-18T10:11:56.786364Z',
    'name': 'Tottenham Hotspur vs. Ajax',
    'parent_id': '929817',
    'short_name': 'TOT vs. AJA',
    'slug': 'tottenham-afc-ajax-2019-04-30',
    'special_rules': null,
    'start_date': '2019-04-30',
    'start_datetime': '2019-04-30T19:00:00Z',
    'state': 'upcoming',
    'type': 'football_match'
  },
  {
    'actual_start_datetime': null,
    'bettable': true,
    'created': '2019-04-18T10:12:03.638683Z',
    'description': null,
    'display_order': 0,
    'end_date': null,
    'full_slug': '/sport/football/europe-uefa-champions-league/2019/05/01/fc-barcelona-vs-liverpool-fc',
    'hidden': false,
    'parent': {
      'name': 'Champions League'
    },
    'id': '1943311',
    'inplay_enabled': true,
    'modified': '2019-04-18T10:12:03.638683Z',
    'name': 'Barcelona vs. Liverpool',
    'parent_id': '929817',
    'short_name': 'BAR vs. LIV',
    'slug': 'fc-barcelona-liverpool-fc-2019-05-01',
    'special_rules': null,
    'start_date': '2019-05-01',
    'start_datetime': '2019-05-01T19:00:00Z',
    'state': 'upcoming',
    'type': 'football_match'
  },
  {
    'actual_start_datetime': null,
    'bettable': true,
    'created': '2019-04-20T10:06:14.610452Z',
    'description': null,
    'display_order': 0,
    'end_date': null,
    'full_slug': '/sport/football/premier-league/2019/05/04/wolverhampton-vs-fulham-fc',
    'hidden': false,
    'parent': {
      'name': 'Premier League'
    },
    'id': '1958062',
    'inplay_enabled': true,
    'modified': '2019-04-20T10:06:14.610452Z',
    'name': 'Wolverhampton Wanderers vs. Fulham',
    'parent_id': '943086',
    'short_name': 'WOL vs. FUL',
    'slug': 'wolverhampton-fulham-fc-2019-05-04',
    'special_rules': null,
    'start_date': '2019-05-04',
    'start_datetime': '2019-05-04T14:00:00Z',
    'state': 'upcoming',
    'type': 'football_match'
  }
];
