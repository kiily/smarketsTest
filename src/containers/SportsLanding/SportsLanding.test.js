import React from 'react';
import { create } from 'react-test-renderer';
import SportsLanding from './SportsLanding';
import { BrowserRouter } from 'react-router-dom';
import { events } from '../../mock-data/event-data';

describe('SportsLanding component', () => {
  it('should match the snapshot when rendered before events load', () => {
    const component = create(<SportsLanding currentSport="football" eventsLoaded={false} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('should match the snapshot when rendered after events load', () => {
    const component = create(<BrowserRouter><SportsLanding currentSport="football" popularEvents={events} eventsLoaded={true} /></BrowserRouter>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
