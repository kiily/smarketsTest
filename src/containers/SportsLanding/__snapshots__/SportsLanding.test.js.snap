// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SportsLanding component should match the snapshot when rendered after events load 1`] = `
Array [
  <div
    className="ContentHeader-content"
  >
    <h1
      className="title"
    >
      Football
       odds
      <button
        className="button-custom "
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-star fa-w-18 "
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </h1>
    <p
      className="description"
    >
      <span>
        Smarkets betting exchange allows you to bet with the best 
        <span>
          Football
        </span>
        odds - thanks to our small margins and industry-low 2% commission - on all tournaments and competitions
        <span>
           including Premier League, La Liga, Europa League and Champions League.
        </span>
      </span>
    </p>
  </div>,
  <ul
    className="EventInfoList-list-container"
  >
    <li
      className="EventInfo-tile tall"
    >
      <a
        href="/event-detail/1925896"
        onClick={[Function]}
      >
        <div
          className="link-overlay"
        />
      </a>
      <div
        className="info-container"
      >
        <div
          className="category-container"
        >
          <span
            className="category-text"
          >
            football
          </span>
          <span
            className="separator"
          >
            &gt;
          </span>
          <span
            className="category-text"
          >
            Championship
          </span>
        </div>
        <div
          className="info-title"
        >
          <div
            className="teams"
          >
            <div
              className="team"
            >
              <span
                className="team-name"
              >
                Millwall
              </span>
            </div>
            <div
              className="team"
            >
              <span
                className="team-name"
              >
                Bristol City
              </span>
            </div>
          </div>
        </div>
        <div
          className="event-date"
        >
          Start date: 
          30/04/2019 19:45
        </div>
      </div>
    </li>
    <li
      className="EventInfo-tile "
    >
      <a
        href="/event-detail/1934371"
        onClick={[Function]}
      >
        <div
          className="link-overlay"
        />
      </a>
      <div
        className="info-container"
      >
        <div
          className="category-container"
        >
          <span
            className="category-text"
          >
            football
          </span>
          <span
            className="separator"
          >
            &gt;
          </span>
          <span
            className="category-text"
          >
            Championship
          </span>
        </div>
        <div
          className="info-title"
        >
          <div
            className="teams"
          >
            <div
              className="team"
            >
              <span
                className="team-name"
              >
                Swansea City
              </span>
            </div>
            <div
              className="team"
            >
              <span
                className="team-name"
              >
                Derby County
              </span>
            </div>
          </div>
        </div>
        <div
          className="event-date"
        >
          Start date: 
          01/05/2019 19:45
        </div>
      </div>
    </li>
    <li
      className="EventInfo-tile "
    >
      <a
        href="/event-detail/1943310"
        onClick={[Function]}
      >
        <div
          className="link-overlay"
        />
      </a>
      <div
        className="info-container"
      >
        <div
          className="category-container"
        >
          <span
            className="category-text"
          >
            football
          </span>
          <span
            className="separator"
          >
            &gt;
          </span>
          <span
            className="category-text"
          >
            Champions League
          </span>
        </div>
        <div
          className="info-title"
        >
          <div
            className="teams"
          >
            <div
              className="team"
            >
              <span
                className="team-name"
              >
                Tottenham Hotspur
              </span>
            </div>
            <div
              className="team"
            >
              <span
                className="team-name"
              >
                Ajax
              </span>
            </div>
          </div>
        </div>
        <div
          className="event-date"
        >
          Start date: 
          30/04/2019 20:00
        </div>
      </div>
    </li>
    <li
      className="EventInfo-tile "
    >
      <a
        href="/event-detail/1943311"
        onClick={[Function]}
      >
        <div
          className="link-overlay"
        />
      </a>
      <div
        className="info-container"
      >
        <div
          className="category-container"
        >
          <span
            className="category-text"
          >
            football
          </span>
          <span
            className="separator"
          >
            &gt;
          </span>
          <span
            className="category-text"
          >
            Champions League
          </span>
        </div>
        <div
          className="info-title"
        >
          <div
            className="teams"
          >
            <div
              className="team"
            >
              <span
                className="team-name"
              >
                Barcelona
              </span>
            </div>
            <div
              className="team"
            >
              <span
                className="team-name"
              >
                Liverpool
              </span>
            </div>
          </div>
        </div>
        <div
          className="event-date"
        >
          Start date: 
          01/05/2019 20:00
        </div>
      </div>
    </li>
    <li
      className="EventInfo-tile "
    >
      <a
        href="/event-detail/1958062"
        onClick={[Function]}
      >
        <div
          className="link-overlay"
        />
      </a>
      <div
        className="info-container"
      >
        <div
          className="category-container"
        >
          <span
            className="category-text"
          >
            football
          </span>
          <span
            className="separator"
          >
            &gt;
          </span>
          <span
            className="category-text"
          >
            Premier League
          </span>
        </div>
        <div
          className="info-title"
        >
          <div
            className="teams"
          >
            <div
              className="team"
            >
              <span
                className="team-name"
              >
                Wolverhampton Wanderers
              </span>
            </div>
            <div
              className="team"
            >
              <span
                className="team-name"
              >
                Fulham
              </span>
            </div>
          </div>
        </div>
        <div
          className="event-date"
        >
          Start date: 
          04/05/2019 15:00
        </div>
      </div>
    </li>
  </ul>,
]
`;

exports[`SportsLanding component should match the snapshot when rendered before events load 1`] = `
Array [
  <div
    className="ContentHeader-content"
  >
    <h1
      className="title"
    >
      Football
       odds
      <button
        className="button-custom "
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-star fa-w-18 "
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </h1>
    <p
      className="description"
    >
      <span>
        Smarkets betting exchange allows you to bet with the best 
        <span>
          Football
        </span>
        odds - thanks to our small margins and industry-low 2% commission - on all tournaments and competitions
        <span>
           including Premier League, La Liga, Europa League and Champions League.
        </span>
      </span>
    </p>
  </div>,
  <div
    className="Spinner-container"
  >
    <span
      className="spinner-text"
    >
      Retrieving popular events
    </span>
    <div
      className="spinner"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>,
]
`;
